erc1155/src/lib.rs

```diff
@@ -70,22 +72,18 @@ impl ERC1155TokenBase for ERC1155Token {
             panic!("ERC1155: accounts and ids length mismatch")
         }
 
-        let mut arr: Vec<BalanceOfBatchReply> = Vec::new();
-
-        for (i, ele) in ids.iter().enumerate() {
-            let account = accounts[i];
-            let amount = self.balance_of(&account, ele);
-
-            let obj = BalanceOfBatchReply {
-                account,
-                id: *ele,
-                amount,
-            };
-
-            arr.push(obj);
-        }
-
-        arr
+        ids.iter()
+            .enumerate()
+            .map(|(idx, x)| BalanceOfBatchReply {
```

Comment:
I'd prefer more usual names.
```suggestion
            .map(|(i, id)| BalanceOfBatchReply {
```

Reply:
Naming changed.